### ISSUE TITLE
vim mode: support ZZ and ZQ for closing pane

### ISSUE THREAD

**Issue by networkhermit** [zed-industries/community#1868]
### Check for existing issues

- [X] Completed

### Describe the feature

Implement ZZ and ZQ for zed vim mode:

- [ ] ZZ: save and close current pane
- [ ] ZQ: force closing pane without confirmation

https://neovim.io/doc/user/editing.html#write-quit

```text
							ZZ

ZZ			Write current file, if modified, and close the current
			window (same as ":x").
			If there are several windows for the current file,
			only the current window is closed.
							ZQ

ZQ			Quit without checking for changes (same as ":q!").
```

**Comment by ConradIrwin** [comment]
This was added in v0.104 which shipped today. Thanks for the suggestion, and please keep them coming!

**Comment by networkhermit** [comment]
Is it possible to skip the `This file contains unsaved edits. Do you want to save it?` prompt?

In vanilla vim and also vscode vim plugin, `ZZ` means `Save` and `ZQ` means `Don't Save` for unsaved content.

**Comment by ConradIrwin** [comment]
Thanks for getting back to me!

I may have erred, but it should be the case that:
* ZZ saves any unsaved changes with no prompt (unless the file has changed on disk, and it will confirm you wanted to overwrite; or if the buffer is Untitled and it'll prompt you for a file name).
* ZQ discards any unsaved changes with no prompt

It sounds like you're seeing something different. Is there a case I missed, or a reproduction step?

**Comment by networkhermit** [comment]
It's weird that I still see the prompt popped up.

Version: Zed 0.105.5

settings.json
```json
{
  "vim_mode": true
}
```

Steps to reproduce:

1. Open an existing file
2. Add some blank lines in the end for test (don't save the buffer)
3. Switch to vim mode and try `ZZ` or `ZQ`
4. A prompt displays `This file contains unsaved edits. Do you want to save it?`

**Comment by ConradIrwin** [comment]
Sorry about that!

I can confirm that there is indeed a bug on v0.105 – a related refactoring means that this is already fixed in 0.106 (currently in preview) which I was testing – my mistake.

The good news is this is fixed in preview, and will likely go out to stable next Wednesday.
